### PR TITLE
cache call to navitia with redis

### PR DESCRIPTION
**Pull request by kinnou02**
On my laptop a call without cache take 75sec, with cache it take 15sec.
The main problem is that the datetime of the request are computed from
'now' so even with the rounding of the since and until, the cache will
be invalidated every hour :(

### CODE REVIEW

File: kirin/gtfs_rt/model_maker.py [hunk 1]
```diff
@@ -123,8 +123,8 @@ def _make_trip_updates(self, input_trip_update, data_time):
     def _get_navitia_vjs(self, trip, data_time):
         vj_source_code = trip.trip_id
 
-        since = data_time - self.period_filter_tolerance
-        until = data_time + self.period_filter_tolerance
+        since = round_datetime(data_time - self.period_filter_tolerance, -1)
```
Comment (xlqian) [inline kirin/gtfs_rt/model_maker.py]: :+1:

File: kirin/gtfs_rt/tasks.py [hunk 2]
```diff
@@ -67,6 +67,9 @@ def gtfs_poller(self, config):
         nav = navitia_wrapper.Navitia(url=config['navitia_url'], token=config['token'])\
                              .instance(config['coverage'])
         nav.timeout = 5
+        pubdate_cache_timeout = app.config.get('NAVITIA_PUBDATE_CACHE_TIMEOUT', 600)
```
Comment (xlqian) [inline kirin/gtfs_rt/tasks.py]: `PUBDATE` i'm not sure if it's a typo?

File: kirin/default_settings.py [hunk 3]
```diff
@@ -51,6 +51,11 @@
 
 ENABLE_RABBITMQ = boolean(os.getenv('KIRIN_ENABLE_RABBITMQ', True))
 
+NAVITIA_QUERY_CACHE_TIMEOUT = int(os.getenv('KIRIN_NAVITIA_QUERY_CACHE_TIMEOUT', 60*60*24))  # in seconds
```
Comment (xlqian) [inline kirin/default_settings.py]: minor: `NAVITIA_QUERY_CACHE_TIMEOUT = int(os.getenv('KIRIN_NAVITIA_QUERY_CACHE_TIMEOUT', timedelta(days=1).total_seconds()))`

File: kirin/gtfs_rt/tasks.py (hunk 2 above)
Comment (kinnou02) [reply kirin/gtfs_rt/tasks.py]: pubdate for publication date, but it's a bit too long for my taste

File: kirin/gtfs_rt/model_maker.py (hunk 1 above)
Comment (pbougue) [reply kirin/gtfs_rt/model_maker.py]: I'd remove that `-1`

Comment (pbougue) [reply kirin/gtfs_rt/model_maker.py]: minor: Also I wouldn't deal this offset into round function (or maybe a round_up/down kind of parameter...). As is, a `floor` function would be a good name

**Pull request merged**